### DISCUSSION TITLE
Docker secrets for default users not loading.

### DISCUSSION THREAD

**Discussion by nblazincic** [rabbitmq/rabbitmq-server#2963] [Other] [answered]
Docker image 3.8.14-management-alpine not loading default user/password  via RABBITMQ_DEFAULT_USER_FILE or PASS variable pointing to docker secrets (ie, /run/secrets/rmq_username)

rolling back to 3.8.5 image works.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This repository is for RabbitMQ server and tier 1 plugins, not the [Docker image](https://github.com/docker-library/rabbitmq). The image has been moving away from using environment variables for a while. These variables were not removed
to my knowledge (our team does not maintain the image but we participate in some decision making around it) but it Is [about to be deprecated](https://github.com/docker-library/rabbitmq/pull/440).

There are several [user management options available](https://www.rabbitmq.com/access-control.html#user-management) in RabbitMQ, with at least two being an option in straight Docker environments. Kubernetes users should use our [cluster Operator](https://www.rabbitmq.com/kubernetes/operator/operator-overview.html) instead of DIY deployments.

**Reply by michaelklishin** [reply]
Err, I really mean to link to [Pre-seeding Users](https://www.rabbitmq.com/access-control.html#seeding) and not the user management section, although that can also be used when building and image and should make consuming secrets from mounted file fairly straightforward. I'd still use either an [encrypted `advanced.config` option](https://www.rabbitmq.com/configure.html#configuration-encryption) or [definition import](https://www.rabbitmq.com/definitions.html#import-on-boot) (which stores password and salt hashes)

**Comment by nblazincic** [comment, marked as answer]
Tnx: posted on https://github.com/docker-library/rabbitmq/issues/479 if somebody find this.

**Reply by michaelklishin** [reply]
@nblazincic you are clinging to a feature that will be deprecated and eventually removed. I don't think Docker maintainers will spend much time investigating this, at least I would not.
